### PR TITLE
TST: GH18498 - Split test_pytables into sub-modules

### LINKED ISSUE #18498: TST: split out test_pytables.py to sub-module of tests
```
(pandas) bash-3.2$ wc -l pandas/tests/io/test_pytables.py 
    5674 pandas/tests/io/test_pytables.py
```
should make a sub-module
pandas/tests/io/pytables

and then split out into various categories of tests, 1 file per.

### PR DESCRIPTION
**Pull request by mikebarkas**
- [x] closes #18498 
- [x] tests moved / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

Moved test classes for pytables into sub-modules.
Moved hdf test data files needed into sub-module.

I did not change any tests, just broke up the classes into separate modules.

### CODE REVIEW

**Comment by mikebarkas** [comment]
PEP8 issue for `common.py` 
`do not use bare except` This is in existing code.
This issue is for moving the tests and not changing test.

File: pandas/tests/io/pytables/common.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
+import pytest
+import os
+import tempfile
+from contextlib import contextmanager
+from distutils.version import LooseVersion
+
+import pandas.util.testing as tm
+
+
+tables = pytest.importorskip('tables')
```
Comment (jreback) [inline pandas/tests/io/pytables/common.py]: if things are not used here, remove them

File: pandas/tests/io/pytables/common.py [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+
+_default_compressor = ('blosc' if LooseVersion(tables.__version__) >=
+                       LooseVersion('2.2') else 'zlib')
+
+
+# contextmanager to ensure the file cleanup
+
+
+def safe_remove(path):
```
Comment (jreback) [inline pandas/tests/io/pytables/common.py]: would like to tun these into fixtures, also this removes the need for the Base class entirely. ok with doing this in another pass though

File: pandas/tests/io/pytables/test_hdf_complex_values.py [hunk 3]
```diff
@@ -0,0 +1,163 @@
+import pytest
+from warnings import catch_warnings
+
```
Comment (jreback) [inline pandas/tests/io/pytables/test_hdf_complex_values.py]: you can just name this test_complex.py

File: pandas/tests/io/pytables/test_hdf_store.py [hunk 4]
```diff
@@ -11,133 +9,34 @@
...
-    @classmethod
-    def teardown_class(cls):
-
-        # Pytables 3.0.0 deprecates lots of things
-        tm.set_testing_mode()
-
-    def setup_method(self, method):
-        self.path = 'tmp.__%s__.h5' % tm.rands(10)
-
-    def teardown_method(self, method):
```
Comment (jreback) [inline pandas/tests/io/pytables/test_hdf_store.py]: the idea is to split this out into many submodules, e.g. 

test_store (construct & instrospect a store)
test_append
test_multiindex
test_remove
test_where
test_select

basically just select similar tests (by name to start is ok) and group them into sub-modules

File: pandas/tests/io/pytables/test_timezones.py [hunk 5]
```diff
@@ -0,0 +1,290 @@
...
+            for i in a.index:
+                a_e = a.loc[i, c]
+                b_e = b.loc[i, c]
+                if not (a_e == b_e and a_e.tz == b_e.tz):
+                    raise AssertionError(
+                        "invalid tz comparison [%s] [%s]" % (a_e, b_e))
+
+    def test_append_with_timezones_dateutil(self):
+
+        from datetime import timedelta
```
Comment (jreback) [inline pandas/tests/io/pytables/test_timezones.py]: move any imports to the top

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by jreback** [comment]
@mikebarkas looks great!

oddly there is 1 less tests here (174 and 1 xfail) vs 175 and 1 xfail on master.

happy to have parameterization of any tests as well.

**Comment by mikebarkas** [comment]
My test build on my fork is failing because  `ImportError: No module named tables`
Looking into it.

I will look for the missing test.

**Comment by mikebarkas** [comment]
This is my missing test
```
def test_longpanel(self):
    pass
```

I will add it back into `test_panel`

**Comment by jreback** [comment]
oh no need to add that back then

File: pandas/tests/io/pytables/test_common.py [hunk 6]
```diff
@@ -0,0 +1,1131 @@
...
+from distutils.version import LooseVersion
+import tempfile
+import datetime
+
+import numpy as np
+import pandas as pd
+from pandas import (Series, DataFrame, date_range, Index, Timestamp,
+                    DatetimeIndex, compat, concat, isna)
+tables = pytest.importorskip('tables')
+from pandas.tests.io.pytables.common import (ensure_clean_store, safe_close,
```
Comment (jreback) [inline pandas/tests/io/pytables/test_common.py]: you can just do a

```
from .common import .....
```

(and do this generally)

File: pandas/tests/io/pytables/test_common.py [hunk 7]
```diff
@@ -0,0 +1,1131 @@
...
+                                             _check_roundtrip, safe_remove,
+                                             _check_roundtrip_table,
+                                             create_tempfile, _maybe_remove,
+                                             ensure_clean_path)
+
+import pandas.util.testing as tm
+import pandas.util._test_decorators as td
+from pandas.util.testing import (assert_panel_equal,
```
Comment (jreback) [inline pandas/tests/io/pytables/test_common.py]: prefer not to import these and just reference with ``tm.``

Comment (jreback) [reply pandas/tests/io/pytables/test_common.py]: also do this generally across all files

File: pandas/tests/io/pytables/test_date.py [hunk 8]
```diff
@@ -0,0 +1,134 @@
+import pytest
+import datetime
+
+import numpy as np
+import pandas as pd
+from pandas import Series, DataFrame, date_range, timedelta_range, bdate_range
```
Comment (jreback) [inline pandas/tests/io/pytables/test_date.py]: can you rename to test_datetime

File: pandas/tests/io/pytables/test_panel.py [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+                                             ensure_clean_store)
+import pandas.util.testing as tm
+from pandas.util.testing import assert_panel_equal
+
+
+def test_wide():
+    with catch_warnings(record=True):
+        wp = tm.makePanel()
+        _check_roundtrip(wp, assert_panel_equal)
+
```
Comment (jreback) [inline pandas/tests/io/pytables/test_panel.py]: there are may more panel tests, can you move here

File: pandas/tests/io/pytables/test_path.py [hunk 10]
```diff
@@ -0,0 +1,163 @@
...
+        df.to_hdf(path, 'df', mode='w', format='table')
+
+        # single
+        store = HDFStore(path)
+        assert 'CLOSED' not in store.info()
+        assert store.is_open
+
+        store.close()
+        assert 'CLOSED' in store.info()
+        assert not store.is_open
```
Comment (jreback) [inline pandas/tests/io/pytables/test_path.py]: these need to go with any other open/close tests, which I think you have in test_common.py now. call this file test_open_close (move this one and other tests)

File: pandas/tests/io/pytables/test_read.py [hunk 11]
```diff
@@ -0,0 +1,270 @@
+import pytest
+from warnings import catch_warnings
+
+import numpy as np
```
Comment (jreback) [inline pandas/tests/io/pytables/test_read.py]: there are many more selection / query type tests that are in test_common. move here

File: pandas/tests/io/pytables/test_read.py [hunk 12]
```diff
@@ -0,0 +1,270 @@
+import pytest
```
Comment (jreback) [inline pandas/tests/io/pytables/test_read.py]: rename this to test_query

File: pandas/tests/io/pytables/test_select.py [hunk 13]
```diff
@@ -0,0 +1,746 @@
+import pytest
+from warnings import catch_warnings
+
+import numpy as np
+from pandas import (Series, DataFrame, date_range, Index, Timestamp,
+                    concat, MultiIndex, bdate_range, Panel)
+from pandas.tests.io.pytables.common import (ensure_clean_store,
```
Comment (jreback) [inline pandas/tests/io/pytables/test_select.py]: consolidate with test_read (and call test_query)

File: pandas/tests/io/pytables/test_start_stop.py [hunk 14]
```diff
@@ -0,0 +1,85 @@
+import pytest
+
+import numpy as np
+import pandas as pd
+from pandas import DataFrame
+from pandas.tests.io.pytables.common import ensure_clean_store
+import pandas.util.testing as tm
+
+
```
Comment (jreback) [inline pandas/tests/io/pytables/test_start_stop.py]: move these to test_select, we don't need a separate file for this

File: pandas/tests/io/pytables/test_store.py [hunk 15]
```diff
@@ -0,0 +1,74 @@
...
+
+import pandas.util.testing as tm
+import pandas.util._test_decorators as td
+
+
+def test_store_hierarchical():
+    index = MultiIndex(levels=[['foo', 'bar', 'baz', 'qux'],
+                               ['one', 'two', 'three']],
+                       labels=[[0, 0, 0, 1, 1, 2, 2, 3, 3, 3],
+                               [0, 1, 2, 0, 1, 1, 2, 0, 1, 2]],
```
Comment (jreback) [inline pandas/tests/io/pytables/test_store.py]: these need to be put in the right place, not a separate file

**Review by jreback — CHANGES_REQUESTED** [review]
a bit too much splitting. idea is to put common files together. tests might be named inconsitenly, but still to group them together, we should end up with only about 5 ish modules here overall.

**Comment by alsobay** [comment]
Hi Mike, are you still working on this? Would be glad to help out or pick it up if you're too busy.

**Comment by jreback** [comment]
closing as stale, happy to take an updated PR of this

**Pull request closed without merging**